#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_lzmasdk_static_lib.dpatch by <gebi@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: support for LZMA compression

@DPATCH@

diff --git a/lzmasdk/C/LzmaUtil/makefile.gcc b/lzmasdk/C/LzmaUtil/makefile.gcc
index f6eec29..44ee18e 100755
--- a/lzmasdk/C/LzmaUtil/makefile.gcc
+++ b/lzmasdk/C/LzmaUtil/makefile.gcc
@@ -1,44 +1,53 @@
-PROG = lzma
-CXX = g++
-LIB =
-RM = rm -f
-CFLAGS = -c -O2 -Wall
-
-OBJS = \
-  LzmaUtil.o \
-  Alloc.o \
-  LzFind.o \
-  LzmaDec.o \
-  LzmaEnc.o \
-  7zFile.o \
-  7zStream.o \
-
-
-all: $(PROG)
-
-$(PROG): $(OBJS)
-	$(CXX) -o $(PROG) $(LDFLAGS) $(OBJS) $(LIB) $(LIB2)
-
-LzmaUtil.o: LzmaUtil.c
-	$(CXX) $(CFLAGS) LzmaUtil.c
-
-Alloc.o: ../Alloc.c
-	$(CXX) $(CFLAGS) ../Alloc.c
-
-LzFind.o: ../LzFind.c
-	$(CXX) $(CFLAGS) ../LzFind.c
-
-LzmaDec.o: ../LzmaDec.c
-	$(CXX) $(CFLAGS) ../LzmaDec.c
-
-LzmaEnc.o: ../LzmaEnc.c
-	$(CXX) $(CFLAGS) ../LzmaEnc.c
-
-7zFile.o: ../7zFile.c
-	$(CXX) $(CFLAGS) ../7zFile.c
-
-7zStream.o: ../7zStream.c
-	$(CXX) $(CFLAGS) ../7zStream.c
-
-clean:
-	-$(RM) $(PROG) $(OBJS)
+PROG = lzma
+CC = gcc
+LIB = liblzma.a
+RM = rm -f
+CFLAGS = -c -O2 -Wall
+AR = ar
+RANLIB = ranlib
+
+OBJS = \
+  Alloc.o \
+  LzFind.o \
+  LzmaDec.o \
+  LzmaEnc.o \
+  LzmaLib.o \
+  7zFile.o \
+  7zStream.o \
+
+all: $(PROG)
+
+$(PROG): LzmaUtil.o $(LIB)
+	$(CC) -o $(PROG) $(LDFLAGS) $< $(LIB)
+
+LzmaUtil.o: LzmaUtil.c
+	$(CC) $(CFLAGS) LzmaUtil.c
+
+$(LIB): $(OBJS)
+	rm -f $@
+	$(AR) rcu $@ $(OBJS)
+	$(RANLIB) $@
+
+Alloc.o: ../Alloc.c
+	$(CC) $(CFLAGS) ../Alloc.c
+
+LzFind.o: ../LzFind.c
+	$(CC) $(CFLAGS) ../LzFind.c
+
+LzmaDec.o: ../LzmaDec.c
+	$(CC) $(CFLAGS) ../LzmaDec.c
+
+LzmaEnc.o: ../LzmaEnc.c
+	$(CC) $(CFLAGS) ../LzmaEnc.c
+
+LzmaLib.o: ../LzmaLib.c
+	$(CC) $(CFLAGS) ../LzmaLib.c
+
+7zFile.o: ../7zFile.c
+	$(CC) $(CFLAGS) ../7zFile.c
+
+7zStream.o: ../7zStream.c
+	$(CC) $(CFLAGS) ../7zStream.c
+
+clean:
+	-$(RM) $(PROG) *.o *.a
